tests/unit/test_utils.py

```diff
@@ -352,6 +354,36 @@ def test_unpack_zip(self, data):
         unzip_file(test_file, self.tempdir)
         self.confirm_files()
 
+    def test_unpack_zip_failure(self):
```

Comment:
I think you should have a success case to parallel your failing case (e.g. to make sure your zip file creation code works). It can be called something like `test_unpack_zip_success` with `make_test_zip_file(test_zip, ['regular_file.txt'])` replacing the creation line you have in the failing case. (And ditto for the tar case.)

Reply:
the successful cases are already covered by existing test_unpack_zip and test_unpack_tgz though they are using a pre-built zip/tar files

Reply:
It’s important to have a “success” case to go along with your failing case because you’re using a different test setup. You want to be sure that your test setup works correctly so you have a baseline to compare to. The existing success case is using a different test setup.